crates/server/src/store/in_memory/authority.rs

```diff
@@ -647,10 +666,25 @@ impl InnerInMemory {
     /// (Re)generates the nsec records, increments the serial number and signs the zone
     #[cfg(feature = "dnssec")]
     #[cfg_attr(docsrs, doc(cfg(feature = "dnssec")))]
-    fn secure_zone_mut(&mut self, origin: &LowerName, dns_class: DNSClass) -> DnsSecResult<()> {
+    fn secure_zone_mut(
+        &mut self,
+        origin: &LowerName,
+        dns_class: DNSClass,
+        nsec3_config: &Nsec3Config,
+    ) -> DnsSecResult<()> {
         // TODO: only call nsec_zone after adds/deletes
         // needs to be called before incrementing the soa serial, to make sure IXFR works properly
-        self.nsec_zone(origin, dns_class);
+        if nsec3_config.enable {
```

Comment:
If you have both NSEC3 and NSEC records in DNS response it is considered invalid per RFC. The way I see it there should be a three-state setting: NSEC3 only, NSEC only, and disable both.

Reply:
I did some local experiments with this idea of a tri-state and it seems the `toml` crate cannot represent `enum`s so something like this:
```rust
pub enum NxProof {
    #[cfg(feature = "dnssec")]
    /// Use NSEC
    Nsec,
    #[cfg(feature = "dnssec")]
    /// Use NSEC3
    Nsec3,
    #[default]
    /// Do not provide non-existence proofs
    None,
}
```
cannot be used. Do you have any suggestions on how to represent this?

Reply:
> both NSEC3 and NSEC records in DNS response it is considered invalid per RFC

Thanks, it's been a while since I've looked at this, good to know.

>  Do you have any suggestions on how to represent this?

@pvdrz, I'm not sure what issue you're running into, here's an example of something representable in TOML: 

https://github.com/hickory-dns/hickory-dns/blob/main/crates/server/src/authority/zone_type.rs#L13-L29

Reply:
Oh it seems I was confused about it. I pushed a commit adding an enum to handle the configuration